File: stress/t/ctl_boolde.f
      PROGRAM CTLBOOLDE
      INTEGER*8 A,B,C,R1,R2,R3,R4,LI,RI
      LOGICAL P,Q,LHS,RHS
      A=3
      B=7
      C=5
      P = (A.LT.B)
      Q = (B.LT.C)
      LHS = .NOT.(P.AND.Q)
      RHS = (.NOT.P).OR.(.NOT.Q)
      IF (LHS) THEN
        LI=1
      ELSE
        LI=0
      END IF
      IF (RHS) THEN
        RI=1
      ELSE
        RI=0
      END IF
      IF (LI.EQ.RI) THEN
        R1=1
      ELSE
        R1=0
      END IF
      WRITE(*,'(A,I0)') 'dm1=',R1
      LHS = .NOT.(P.OR.Q)
      RHS = (.NOT.P).AND.(.NOT.Q)
      IF (LHS) THEN
        LI=1
      ELSE
        LI=0
      END IF
      IF (RHS) THEN
        RI=1
      ELSE
        RI=0
      END IF
      IF (LI.EQ.RI) THEN
        R2=1
      ELSE
        R2=0
      END IF
      WRITE(*,'(A,I0)') 'dm2=',R2
      IF (P.AND.Q) THEN
        R3=1
      ELSE
        R3=0
      END IF
      IF (P.OR.Q) THEN
        R4=1
      ELSE
        R4=0
      END IF
      WRITE(*,'(A,I0)') 'r3=',R3
      WRITE(*,'(A,I0)') 'r4=',R4
      IF (.NOT.P) THEN
        R1=1
      ELSE
        R1=0
      END IF
      IF ((.NOT.(A.EQ.B)).AND.(C.NE.A)) THEN
        R2=1
      ELSE
        R2=0
      END IF
      WRITE(*,'(A,I0)') 'r5=',R1
      WRITE(*,'(A,I0)') 'r6=',R2
      END

File: stress/t/ctl_boolde.c
#include <stdio.h>
#include <stdint.h>
int main(void){
  int64_t a=3,b=7,c=5,r1=0,r2=0,r3=0,r4=0,li=0,ri=0;
  int p,q,lhs,rhs;
  p=(a<b);
  q=(b<c);
  lhs=!(p && q);
  rhs=(!p) || (!q);
  li = lhs ? 1 : 0;
  ri = rhs ? 1 : 0;
  r1 = (li==ri) ? 1 : 0;
  printf("dm1="); printf("%ld",(long)r1); printf("\n");
  lhs=!(p || q);
  rhs=(!p) && (!q);
  li = lhs ? 1 : 0;
  ri = rhs ? 1 : 0;
  r2 = (li==ri) ? 1 : 0;
  printf("dm2="); printf("%ld",(long)r2); printf("\n");
  r3 = (p && q) ? 1 : 0;
  r4 = (p || q) ? 1 : 0;
  printf("r3="); printf("%ld",(long)r3); printf("\n");
  printf("r4="); printf("%ld",(long)r4); printf("\n");
  r1 = (!p) ? 1 : 0;
  r2 = (!(a==b) && (c!=a)) ? 1 : 0;
  printf("r5="); printf("%ld",(long)r1); printf("\n");
  printf("r6="); printf("%ld",(long)r2); printf("\n");
  return 0;
}
